pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -107,22 +109,32 @@ func (m *kubeGenericRuntimeManager) generateLinuxContainerConfig(container *v1.C
 			unified[cm.MemoryMin] = strconv.FormatInt(memoryRequest, 10)
 		}
 
-		// If container sets limits.memory, we set memory.high=pod.spec.containers[i].resources.limits[memory] * memory_throttling_factor
-		// for container level cgroup if memory.high>memory.min.
-		// If container doesn't set limits.memory, we set memory.high=node_allocatable_memory * memory_throttling_factor
-		// for container level cgroup.
-		memoryHigh := int64(0)
-		if memoryLimit != 0 {
-			memoryHigh = int64(float64(memoryLimit) * m.memoryThrottlingFactor)
-		} else {
-			allocatable := m.getNodeAllocatable()
-			allocatableMemory, ok := allocatable[v1.ResourceMemory]
-			if ok && allocatableMemory.Value() > 0 {
-				memoryHigh = int64(float64(allocatableMemory.Value()) * m.memoryThrottlingFactor)
+		// Guaranteed pods by their QoS definition require memory requests=memory limits and are not overcommitted.
+		// Hence MemoryQoS feature is disabled on those pods by not setting memory.high.
+		if memoryRequest != memoryLimit {
```

Comment:
nit: It's worth to call out here that "officially" to be in guaranteed QoS cpu request must equal cpu limit AND memory request must equal to memory limit. I agree that for this feature, checking cpu is not really important, so maybe we need to have an exception here and say we only do this for pods where memoryRequest != memory limit, or "their memory is not guaranteed since they are burstable from memory perspective", but it's worth to mention.

Reply:
I updated the comment here.